src/Compilers/CSharp/Portable/BoundTree/BoundNode.cs

```diff
@@ -170,29 +170,19 @@ public void ResetCompilerGenerated(bool newCompilerGenerated)
         /// <summary>
         /// Top level nullability for the node. This should not be used by flow analysis.
         /// </summary>
+        [DebuggerHidden]
         protected NullabilityInfo TopLevelNullability
         {
             get
             {
 #if DEBUG
                 _attributes |= BoundNodeAttributes.WasTopLevelNullabilityChecked;
 #endif
-                if ((_attributes & BoundNodeAttributes.TopLevelAnnotationMask) == 0)
-                {
-                    return default;
-                }
-
-                var annotation = (_attributes & BoundNodeAttributes.TopLevelAnnotationMask) switch
-                {
-                    BoundNodeAttributes.TopLevelAnnotated => CodeAnalysis.NullableAnnotation.Annotated,
-                    BoundNodeAttributes.TopLevelNotAnnotated => CodeAnalysis.NullableAnnotation.NotAnnotated,
-                    BoundNodeAttributes.TopLevelDisabled => CodeAnalysis.NullableAnnotation.Disabled,
-                    var mask => throw ExceptionUtilities.UnexpectedValue(mask)
-                };
 
-                var flowState = (_attributes & BoundNodeAttributes.TopLevelFlowStateMaybeNull) == 0 ? CodeAnalysis.NullableFlowState.NotNull : CodeAnalysis.NullableFlowState.MaybeNull;
-
-                return new NullabilityInfo(annotation, flowState);
+                // This is broken out into a separate property so the debugger can display the
+                // top level nullability without setting the _attributes flag and interferring
+                // with the normal operation of tests.
+                return TopLevelNullabilityDebuggerDisplay;
```

Comment:
>TopLevelNullabilityDebuggerDisplay [](start = 23, length = 34)

This property is not just for debugger display, it is the core value. Perhaps name it `TopLevelNullabilityCore` instead.